build-system/tasks/generate-runner.js

```diff
@@ -109,13 +109,19 @@ async function generateRunner(subDir) {
   const runnerJarDir = `${runnerDistDir}/${subDir}`;
   writeGeneratedAtCommitFile(runnerJarDir);
   const result = getOutput(generateCmd);
-  if (result.stderr) {
+  if (0 !== result.status) {
     log(
       red('ERROR:'),
       'Could not generate custom closure compiler',
       cyan(`${runnerJarDir}/runner.jar`)
     );
-    console.error(red(result.stdout), red(result.stderr));
+    console.error(
+      red(result.stdout),
+      red(result.stderr),
+      red(
+        'See instructions for setting up Java at https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server'
+      )
```

Comment:
I just tested this by injecting a genuine error in runner generation, and have an improvement to suggest in the messaging.

- Add a constant at the top of the file:
```js
const setupInstructionsUrl = 'https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server';
```

- Replace line 118 in the original code with this block:
```js
    console.error(red(result.stdout), red(result.stderr));
    log(
      green('INFO:'),
      'If the errors above are in java execution, see',
      cyan(`${setupInstructionsUrl}`)
    );
```